src/lib/app-state-hoc.jsx

```diff
@@ -37,8 +40,22 @@ const AppStateHOC = function (WrappedComponent) {
             } else {
                 intl = intlInitialState;
             }
+            if (props.isPlayerOnly || props.isFullScreen) {
+                mode = {
+                    isFullScreen: props.isFullScreen || false,
+                    isPlayerOnly: props.isPlayerOnly || false
+                };
+            } else {
+                mode = modeInitialState;
+            }
 
-            this.store = createStore(reducer, intl, enhancer);
+            this.store = createStore(
+                reducer,
+                {
+                    intl: intl,
+                    mode: mode
+                },
```

Comment:
I'm surprised that this works since previously the initial state was just the initial state for intl, but now it's an object containing keys for each state.  What you have now actually makes more sense to me — maybe it just wasn't working all the way before?

Reply:
It was working because the initial state includes the `intl` key.